### PR TITLE
Remove redundant version checks

### PR DESCRIPTION
**Pull request by MGaetan89**
Now that the minimum SDK version of Material Components is 19, there was a lot of code that could be simplified or removed.

### CODE REVIEW

File: lib/java/com/google/android/material/badge/BadgeUtils.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
 @ExperimentalBadgeUtils
 public class BadgeUtils {
 
-  public static final boolean USE_COMPAT_PARENT = VERSION.SDK_INT < VERSION_CODES.JELLY_BEAN_MR2;
+  public static final boolean USE_COMPAT_PARENT = false;
```
Comment (MGaetan89) [inline lib/java/com/google/android/material/badge/BadgeUtils.java:50]: I didn't know how to proceed here: `BadgeUtils` is a public class, and so is this `USE_COMPAT_PARENT` constant.
Can I simply remove it? Or should I deprecate it for removal in a future release?

**Comment by MGaetan89** [comment]
I've fixed the conflict in `CarouselLayoutManagerRtlTest`

**Comment by MGaetan89** [comment]
@hunterstich I've updated the PR and removed some additional legacy code.

**Comment by leticiarossi** [comment]
Thank you for working on this! I removed the change on the Badge file since it'd be best to refactor badge instead of just having a variable set to false, so that should be done in a separate change.

**Comment by MGaetan89** [comment]
> Thank you for working on this! I removed the change on the Badge file since it'd be best to refactor badge instead of just having a variable set to false, so that should be done in a separate change.

Are you open for a PR to remove `BadgeUtils.USE_COMPAT_PARENT`?

**Comment by leticiarossi** [comment]
sure if you'd like to contribute feel free to do so and we'll happily review it! :)

**Comment by MGaetan89** [comment]
I'll prepare a PR then 🙂 
Just to be sure that we're aligned: the goal is to remove `BadgeUtils.USE_COMPAT_PARENT` and every it's used, correct?

**Comment by leticiarossi** [comment]
that's right, we should remove the checks for `BadgeUtils.USE_COMPAT_PARENT` similar to how you removed the checks for API <= 19

**Comment by MGaetan89** [comment]
Ok, https://github.com/material-components/material-components-android/pull/4170 is ready when you have time to have a look 🙂

**Pull request closed without merging**